packaging/setup/ovirt_engine_setup/keycloak/constants.py

```diff
@@ -66,4 +67,22 @@ class FileLocations(oesetupcons.FileLocations):
         '12-setup-keycloak.conf'
     )
 
+    KEYCLOAK_ADD_USER_SCRIPT = os.path.join(
+        oesetupcommconfig.JBOSS_HOME,
+        "bin",
+        "add-user-keycloak.sh",
+    )
+
+    OVIRT_ENGINE_CONFIG_DIR = os.path.join(
+        oenginecons.FileLocations.OVIRT_ENGINE_DATADIR,
+        "services",
+        "ovirt-engine"
+    )
+
+    KEYCLOAK_ADD_INITIAL_ADMIN_FILE = os.path.join(
+        OVIRT_ENGINE_CONFIG_DIR,
+        "keycloak-add-user.json",
```

Comment:
Where is this json comes from?

Reply:
from keycloak 'add-user-keycloak.sh'  command line tool. It's default & hardcoded.

Reply:
Hmm, I haven't found tthis json file in none of currently existting RPMs:

https://download.copr.fedorainfracloud.org/results/ovirt/ovirt-master-snapshot/centos-stream-8-x86_64/03080336-ovirt-engine-keycloak/

Reply:
And I can't see it even in this patch, so am I missing something?

Reply:
that json is the result file from running:
```add-user-keycloak.sh -r master -u admin -p <passwd> -sc /path/to/wildfly/server/config/dir  ```
```add-user-keycloak.sh``` is part of keycloak overlay package

Reply:
Ah OK, now I understand. So Keycloak need that JSON file forever? If so, does it contain any sensitive information? If so, we should probably change ownership to ovirt:ovirt and mode to 0600

Reply:
Technically it should live until systemctl start ovirt-engine. Even more, after this file is copied to jbossRuntime and processed by keycloak module on its loading phase, keycloak then removes it.
I can remove it from /usr/share/ovirt-engine/services/ovirt-engine/   on startup, I have even considered this option, but eventually decided to wait with it until the review. I see one issue though -  ie. if ovirt  ovirt engine service cannot be started (because of whatever reason) then we will loose that configuration
So 0600 mode seems like a good trade off.

Reply:
+1

Reply:
done